syntax = "proto2";

message PositionState {
    optional float x = 1;
    optional float y = 2;
}

enum EntityType {
    ENTITY = 0;
    SHIP = 1;
};

message EntityState {
    required int32 id = 1;
    optional PositionState position = 2;
    required EntityType type = 3;
}

message GameState {
    repeated EntityState entities = 1;
}